From 62bbc2b2bdeb5e79eaf3b315ad22cf18c3988b27 Mon Sep 17 00:00:00 2001
From: Amr Bekhit <amrbekhit@gmail.com>
Date: Mon, 7 Oct 2019 12:16:03 +0300
Subject: [PATCH] Disable addition of simple-framebuffer by U-boot

In order to enable HDMI support for the Raspberry Pi, U-boot adds a
simple-framebuffer node to the device tree. However, this then prevents
the kernel from being able to correctly initialise the HDMI (which results
in a corrupted terminal), which subsequently prevents the camera from working.
Disabling HDMI support in U-boot resolves this issue. The U-boot console is
still available over the serial port.

Signed-off-by: Amr Bekhit <amrbekhit@gmail.com>
---
 board/raspberrypi/rpi/rpi.c | 7 -------
 include/configs/rpi.h       | 3 ---
 2 files changed, 10 deletions(-)

diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
index dfc96b471e..22211e8b0c 100644
--- a/board/raspberrypi/rpi/rpi.c
+++ b/board/raspberrypi/rpi/rpi.c
@@ -505,13 +505,6 @@ void *board_fdt_blob_setup(void)
 
 int ft_board_setup(void *blob, bd_t *bd)
 {
-	/*
-	 * For now, we simply always add the simplefb DT node. Later, we
-	 * should be more intelligent, and e.g. only do this if no enabled DT
-	 * node exists for the "real" graphics driver.
-	 */
-	lcd_dt_simplefb_add_node(blob);
-
 #ifdef CONFIG_EFI_LOADER
 	/* Reserve the spin table */
 	efi_add_memory_map(0, 1, EFI_RESERVED_MEMORY_TYPE, 0);
diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 68790a577f..f44d2f5f22 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -59,9 +59,6 @@
 /* Devices */
 /* GPIO */
 #define CONFIG_BCM2835_GPIO
-/* LCD */
-#define CONFIG_LCD_DT_SIMPLEFB
-#define CONFIG_VIDEO_BCM2835
 
 #ifdef CONFIG_CMD_USB
 #define CONFIG_TFTP_TSIZE
-- 
2.17.1

